lib/spack/spack/reporters/cdash.py

```diff
@@ -194,11 +214,34 @@ def clean_log_event(event):
                 t = env.get_template(phase_template)
                 f.write(t.render(report_data))
             self.upload(phase_report)
+
+    def build_report(self, directory_name, input_data):
+        # Do an initial scan to determine if we are generating reports for more
+        # than one package. When we're only reporting on a single package we
+        # do not explicitly include the package's name in the CDash build name.
```

Comment:
In my case, since I don't rely on the `install_command` but rather supply the build name I want via the `--cdash-build` argument, the naming here doesn't affect me... unless somehow one of my `spack install ...` commands results in multiple builds.  I guess it should only happen in either of two cases: 1) the command line includes multiple specs to install, or 2) the command line defines only a single spec, but the dependencies need to be installed?  If it's true that multiple builds should only happen in one of those two cases, then I guess it shouldn't ever happen in our release pipeline, as I expect all my dependencies to be available in the buildcache.

Reply:
> the command line defines only a single spec, but the dependencies need to be installed

This is the main use case where I foresee multiple CDash builds getting created.

So you're right, as long as your dependencies are available in the buildcache, we should have a one-to-one relationship between CDash builds and calls to `spack install`.